stackit/internal/services/mariadb/instance/datasource.go

```diff
@@ -168,9 +168,8 @@ func (r *instanceDataSource) Read(ctx context.Context, req datasource.ReadReques
 	instanceResp, err := r.client.GetInstance(ctx, projectId, instanceId).Execute()
 	if err != nil {
 		oapiErr, ok := err.(*oapierror.GenericOpenAPIError) //nolint:errorlint //complaining that error.As should be used to catch wrapped errors, but this error should not be wrapped
-		if ok && (oapiErr.StatusCode == http.StatusNotFound) || (oapiErr.StatusCode == http.StatusGone) {
+		if ok && (oapiErr.StatusCode == http.StatusNotFound || oapiErr.StatusCode == http.StatusGone) {
 			resp.State.RemoveResource(ctx)
```

Comment:
We don't need to remove it from the state, the fix in this PR doesn't apply to datasources in general. 
If the user is reading something that got deleted, it should receive an error and then the user needs to remove the datasource block from his configuration. 
Removing it from the state will not make a difference

Reply:
Discussed offline, we should still throw an error but removing it from the state is also a valid use case so the user doesn't have do to it manually